aws/resource_aws_codebuild_project.go

```diff
@@ -95,6 +95,21 @@ func resourceAwsCodeBuildProject() *schema.Resource {
 										Type:     schema.TypeString,
 										Required: true,
 									},
+									"type": {
+										Type:     schema.TypeString,
+										Optional: true,
+										ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+											value := v.(string)
+											validType := []string{"PARAMETER_STORE", "PLAINTEXT"}
+											for _, str := range validType {
+												if value == str {
+													return
+												}
+											}
+											errors = append(errors, fmt.Errorf("expected %s to be one of %v, got %s", k, validType, value))
+											return
+										},
```

Comment:
Nitpick: Do you mind replacing this with a standard helper we have in the upstream library and constants from the SDK? i.e.

```go
ValidateFunc: validation.StringInSlice([]string{
  codebuild.EnvironmentVariableTypeParameterStore,
  codebuild.EnvironmentVariableTypePlaintext,
}, true),
```